Task12-whack-a-mole/src/game.js

```diff
@@ -0,0 +1,131 @@
+const game = new Game();
+game.initialize();
+
+function Game() {
+    const holes = document.querySelectorAll('.hole');
+    const scoreBoard = document.querySelector('.score');
+    const game = document.querySelector('.game');
+    const moles = document.querySelectorAll('.mole');
+    const scoreCard = document.querySelector('.scores');
+    const inputName = document.querySelector('#pname');
+    const levelText = document.querySelector('.level');
+    const timeText = document.querySelector('.time');
+
+    const minPeepTime = 200;
+    let maxPeepTime = 1000;
+    const failAudio = "kick";
+    const hitAudio = "hihat"
+    const elementToHit = "mole";
+    const gameTime = 10000;
+
+    let players = [];
+    let lastHole;
+    let timeUp = false;
+    let score = 0;
+    let playerName = "";
+    let level = 1;
+
+    this.initialize = () => {
+        game.addEventListener('click', this.bonk);
+    }
+
+    this.randomTime = (min, max) => {
+        return Math.round(Math.random() * (max - min) + min);
+    }
+
+    this.randomHole = holes => {
+        const idx = Math.floor(Math.random() * holes.length);
+        const hole = holes[idx];
+        if (hole === lastHole) return this.randomHole(holes);
+        lastHole = hole;
+        return hole;
+    }
+
+    this.peep = () => {
+        const time = this.randomTime(minPeepTime, maxPeepTime);
+        const hole = this.randomHole(holes);
+        hole.classList.add('up');
+        setTimeout(() => {
+            hole.classList.remove('up');
+            if (!timeUp) this.peep();
+        }, time);
+    }
+
+    this.updateTime = () => {
+        setTimeout(() => {
+            if (!timeUp) {
+                console.log(timeText.textContent);
+                timeText.textContent = timeText.textContent - 1;
+                this.updateTime();
+            }
+        }, 1000)
+    }
+
+    this.startGame = () => {
+        playerName = (inputName.value == "") ? "Anonymous" : inputName.value;
+
+        timeText.textContent = (gameTime / 1000);
+        scoreBoard.textContent = 0;
+        score = 0;
+        levelText.textContent = 1;
+        level = 1;
+
+        timeUp = false;
+        this.peep();
+        this.updateTime();
+        setTimeout(() => this.finishGame(), gameTime)
+    }
+
+    this.sortPlayers = () => {
+        players.sort((player1, player2) => {
+            if (player1.score < player2.score) {
+                return 1;
+            }
+            else if (player1.score > player2.score) {
+                return -1;
+            }
+            return 0;
+        });
```

Comment:
When you sort an array and the compare function involves numbers, you can simply do:
`players.sort((player1, player2) => player1.score - player2.score) `
It would will you the same result